docs/tutorial/03-resources.md

```diff
@@ -19,581 +19,615 @@ socialImageTitle: Cadence Resources
+```cadence
+import HelloResource from 0x06
 
 transaction {
+    var result: String
+    // Other code...
+}
+```
 
-    /// In this prepare block, we have to load a value from storage
-    /// in addition to saving it, so we also need the `LoadValue` entitlement
-    /// which additionally allows loading values from storage
-    prepare(acct: auth(LoadValue, SaveValue) &Account) {
+You'll get an error: `missing initialization of field `result` in type `Transaction`. not initialized`
 
-        // Load the resource from storage, specifying the type to load it as
-        // and the path where it is stored
-        let helloResource <- acct.storage.load<@HelloWorld.HelloAsset>(from: /storage/HelloAssetTutorial)
-            ?? panic("The signer does not have the HelloAsset resource stored at /storage/HelloAssetTutorial. Run the `Create Hello` Transaction again to store the resource")
+In transactions, variables at the `transaction` level must be initialized in the `prepare` phase.
 
-        // log the hello world message
-        log(helloResource.hello())
+:::info[Action]
 
-        // Put the resource back in storage at the same spot
-        acct.storage.save(<-helloResource, to: /storage/HelloAssetTutorial)
-    }
-}
+Initialize the `result` message and create a constant for the storage path.
+
+:::
+
+```cadence
+self.result = "Saved Hello Resource to account."
+let storagePath = /storage/HelloAssetTutorial
 ```
 
-Here's what this transaction does:
+:::warning
 
-1. Import the `HelloWorld` definitions from account `0x06`
-2. Moves the `HelloAsset` object from storage to `helloResource` with the move operator
-  and the `load` function from the [account storage API](../language/accounts/storage.mdx)
-3. Calls the `hello()` function of the `HelloAsset` resource stored in `helloResource` and logs the result
-4. Saves the resource in the account that we originally moved it from at the path `/storage/HelloAssetTutorial`
+In Cadence, storage paths are a type.  They are **not** `Strings` and are not enclosed by quotes.
 
-We're going to be using the `prepare` phase again to load the resource
-using the [reference to the account](../language/accounts/index.mdx) that is passed in.
+:::
 
-Let's go over the transaction in more detail.
+One way to check whether or not a storage path has an object in it is to use the built-in [`storage.check`] function with the type and path.  If the result is `true`, then there is an object in account storage that matches the type requested.  If it's `false`, there is not.
 
-#### Loads the `HelloAsset` resource from storage
+```warning
 
-To remove an object from storage, we use the `load` method from the [account storage API](../language/accounts/storage.mdx)
+A response of `false` does **not** mean the location is empty.  If you ask for an apple and the location contains an orange, this function will return `false`.
+
+This is not likely to occur, but is theoretically possible.
```

Comment:
```suggestion
This is not likely to occur because projects are encouraged to create storage and public paths that are very unique, but is theoretically possible if projects don't follow this best practice or if there is a malicious app that tries to store things in other projects' paths.
```